openff/toolkit/topology/molecule.py

```diff
@@ -3880,52 +3866,22 @@ def _add_chemical_info_to_networkx_graph(
 
         pdb = PDBFile(file_path)
 
-        omm_topology_G = Topology._openmm_topology_to_networkx(pdb.topology)
-
-        omm_topology_G = _add_chemical_info_to_networkx_graph(
-            substructure_dictionary, omm_topology_G, toolkit_registry=toolkit_registry
+        rdkit_mol = openmm_to_rdkit(pdb)
+        rdkit_mol = _add_chemical_info(
+            substructure_dictionary, rdkit_mol, toolkit_registry=toolkit_registry
         )
-
-        offmol = Molecule()
-
-        for node_idx, node_data in omm_topology_G.nodes.items():
-            offmol.add_atom(
-                node_data["atomic_number"],
-                int(node_data["formal_charge"]),
-                False,
-                name=node_data["atom_name"],
-                metadata={
-                    "residue_name": node_data["residue_name"],
-                    "residue_number": node_data["residue_id"],
-                    "chain_id": node_data["chain_id"],
-                },
-            )
-
-        for edge, edge_data in omm_topology_G.edges.items():
-            offmol.add_bond(edge[0], edge[1], edge_data["bond_order"], False)
-
-        coords = unit.Quantity(
-            np.array(
-                [
-                    [*vec3.value_in_unit(openmm_unit.angstrom)]
-                    for vec3 in pdb.getPositions()
-                ]
-            ),
-            unit.angstrom,
-        )
-
-        offmol.add_conformer(coords)
-
-        offmol_w_stereo_and_aro = toolkit_registry.call(
-            "_assign_aromaticity_and_stereo_from_3d", offmol
-        )
-
-        assert offmol_w_stereo_and_aro.n_atoms == offmol.n_atoms
-        for stereo_aro_atom, atom in zip(offmol_w_stereo_and_aro.atoms, offmol.atoms):
-            atom.stereochemistry = stereo_aro_atom.stereochemistry
-            atom._is_aromatic = stereo_aro_atom.is_aromatic
-        for stereo_aro_bond, bond in zip(offmol_w_stereo_and_aro.bonds, offmol.bonds):
-            bond._is_aromatic = stereo_aro_bond.is_aromatic
+        # TODO: I think we've copied this across from the SMARTS patterns, unless I've confused R/S and CCW/CW again
+        # Chem.AssignStereochemistryFrom3D(rdkit_mol)
+
+        offmol = Molecule.from_rdkit(rdkit_mol)
+        for i, atom in enumerate(pdb.topology.atoms()):
+            offmol.atoms[i].name = atom.name
+            # shh, don't tell Jeff
```

Comment:
This comment is referring to directly editing the private variable, I wasn't sure if there was a proper way to assign metadata

Reply:
Ah, gotcha - Thanks for the heads up :-)